applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -0,0 +1,265 @@
+}
+
+// void DistanceModificationProcess::ExecuteInitialize() {
+
+//     KRATOS_TRY;
+
+//     // Continuous distance field required variables check
+//     if (mContinuousDistance){
+//         const auto& r_node = *mrModelPart.NodesBegin();
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_H, r_node);
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
+//     }
+
+//     KRATOS_CATCH("");
+// }
+
+void MassConservationCheckProcess::ExecuteFinalizeSolutionStep()
+{
+  
+    if ( this->mIsSwitchedOn ){
+            double negVol = 0.0;
+            double posVol = 0.0;
+
+        if ( mCompareToInitial && mCounter == 1){
+            // getting initial values and storing as reference
+            this->ComputeVolumesOfFluids( posVol, negVol );
+            mInitialPositiveVolume = posVol;
+            mInitialNegativeVolume = negVol;
+        }
+
+        if ( mCounter % this->mMassComputationFreq == 0){
+            // writing an output at a given frequncy
+            this->ComputeVolumesOfFluids( posVol, negVol );
+            std::cout << " --- Volume Checking Process --- " << std::endl;
+            std::cout << "   positive Volume = " << posVol << std::endl;
+            if ( mCompareToInitial ){ 
+                std::cout << "   ( " << posVol / mInitialPositiveVolume * 100.0 << "% of initial )" << std::endl; 
+            }
+            std::cout << "   negative Volume = " << negVol << std::endl;
+            if ( mCompareToInitial ){ 
+                std::cout << "   ( " << negVol / mInitialNegativeVolume * 100.0 << "% of initial )" << std::endl; 
+            }
+            std::cout << " --- --- --- --- --- --- --- --- " << std::endl;
+
+            if (mWriteToLogFile){
+                std::ofstream myfile;
+                myfile.open ("volumeLog.txt", std::ios::out | std::ios::app);
+                if ( myfile.is_open() ){
+
+                    myfile << mCounter << "    ";
+                    myfile << posVol << "    ";
+                    if ( mCompareToInitial ){ 
+                        myfile << posVol / mInitialPositiveVolume * 100.0 << "    "; 
+                    }
+                    myfile << negVol << "    ";
+                    if ( mCompareToInitial ){ 
+                        myfile << negVol / mInitialNegativeVolume * 100.0 << "\n"; 
+                    } 
+                    myfile.close();
+                }
+            }
+        }
+
+        mCounter++;
+    } else {
+        return;
+    }
+};
+
+
+/* Private functions ****************************************************/
+
+void MassConservationCheckProcess::ComputeVolumesOfFluids( double &positiveVolume, double &negativeVolume )
+{
+    // useless containers
+    Matrix rShapeFunctionsPos, rShapeFunctionsNeg;
+    GeometryType::ShapeFunctionsGradientsType rShapeDerivativesPos, rShapeDerivativesNeg;
+    Vector null3(3, 0.0);
+    Vector null4(4, 0.0);
+
+    // abbreviation of variable names
```

Comment:
Use the function arguments. If you consider them to be too long use an alternative ones.

Reply:
Yes, that is better.